docs/development/making_pr.rst

```diff
@@ -0,0 +1,166 @@
+
+  git checkout -B change-one
+
+
+Make Your Changes and Commit
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now enter the directory of your fork amd make changes as you wish.
+Run tests for the changes you have made.
+
+If you create a new file, remember to add it with :code:`git add`.
+
+.. code:: sh
+
+  git add <new-file>
+
+Commit your changes, adding a description of what was added. If you’re not
+used to Git, the simplest way is to commit all modified files and add a
+description message of your changes in a single command like this:
+
+.. code:: sh
+
+  git commit -a -m "<Description of changes made>"
+
+Merging
+~~~~~~~~
+
+Switch to the :code:`develop` branch and merge the changes from
+:code:`change-one` branch to the :code:`develop` branch. Delete the
+:code:`change-one` branch, if you do not plan to use it anymore.
+
+.. code:: sh
+
+  git checkout develop # switching to develop branch
+  git merge change-one # merging the changes from change-one branch
+  git branch -d change-one # deleting the change-one branch
+
+:code:`git merge` incorporates changes from the named commits (since the time their histories
+diverged from the current branch) into the current branch. This command is
+used by `git pull` to incorporate changes from another repository and can be
+used by hand to merge changes from one branch into another.
+
+Assume the following history exists and the current branch you are on
+is :code:`develop`:
+
+.. code:: sh
+
+        A---B---C change-one
+       /
+  D---E---F---G develop
+
+Then :code:`git merge change-one` will replay the changes made on the
+:code:`change-one` branch since it diverged from :code:`develop` (i.e., E)
+until its current commit (C) on top of
+:code:`develop`, and record the result in a new commit along with the names of
+the two parent commits and a log message from the user describing the changes.
+
+.. code:: sh
+
+	  A---B---C change-one
+	 /         \
+    D---E---F---G---H develop
+
+.. note::
+  It is not necessary to merge :code:`change-one` branch into
+  :code:`develop` branch in order to open a pull request. It is perfectly fine
+  to open a pull request from the :code:`change-one` branch.
+
+Rebasing
+~~~~~~~~~
+
+Rebasing is the process of moving or combining a sequence of commits to a new
+base commit. Rebasing is most useful and easily visualized in the context of a
+feature branching workflow.
+
+Assume the following history exists:
+
+.. code:: sh
+
+          A---B---C change-one
+         /
+    D---E---F---G develop
+
+From this point, the result of either of the following commands:
+
+.. code:: sh
+
+  git rebase develop
+  git rebase develop change-one
+
+would be:
+
+.. code:: sh
+
+                    A`--B`--C` change-one
+                 /
+    D---E---F---G develop
+
+.. note::
+  :code:`A` and :code:`A`` represents the same set of changes, but have
+  different committer information. 
+
+.. warning::
+  The golden rule of git rebase is to never use it on public branches.
```

Comment:
Generally you are right here - but we _do_ ask people to interactively rebase to squash commits - and this means they have to force push :) So we should remove this warning to avoid confusion.